app/backend/src/core/controllers/pricing/PriceUpdaterController.ts

```diff
@@ -0,0 +1,124 @@
+import { PriceRepository } from "../../../database/repositories/PricingRepository";
+import { WhitelistedAsset } from "../../../core/clients/coincap/assets/types";
+import whitelisted from "../../clients/coincap/assets/whitelisted.json";
+import CoinCapClient from "./../../clients/coincap/CoinCapClient";
+import { UnixTime } from "../../../core/types/UnixTime";
+import Logger from "../../../tools/Logger";
+
+export class PriceUpdaterController {
+  private readonly coinCapClient: CoinCapClient;
+  private readonly priceRepository: PriceRepository;
+  private readonly whitelistedAssets: WhitelistedAsset[];
+  private readonly logger: Logger;
+
+  private priceIntervalMinutes: number;
+  private apiIntervalLimitMinutes: number;
+
+  constructor(
+    coinCapClient: CoinCapClient,
+    priceRepository: PriceRepository,
+    logger: Logger,
+  ) {
+    this.coinCapClient = coinCapClient;
+    this.priceRepository = priceRepository;
+    this.logger = logger;
+
+    this.whitelistedAssets = whitelisted;
+    this.priceIntervalMinutes = 15;
+    this.apiIntervalLimitMinutes = 7;
+  }
+
+  async start() {
+    await this.updatePricesForAllAssets();
+  }
+
+  private async updatePricesForAllAssets() {
+    const latestBoundaries = await this.priceRepository.findLatestByToken(
+      UnixTime.now().add(-30, "days"),
+    );
+    for (const asset of this.whitelistedAssets) {
+      const latestBoundary: UnixTime | undefined = latestBoundaries.get(
+        asset.coincap_asset_id,
+      );
+      await this.updatePricesForAsset(asset, latestBoundary);
+    }
+  }
+
+  private async updatePricesForAsset(
+    asset: WhitelistedAsset,
+    latestBoundary?: UnixTime,
+  ) {
+    // It will fetch the last 30 days of data if there is no latestBoundary, ie. new assets
+    // If the latestBoundary is older than the priceIntervalMinutes (15 min refresh rate), it will fetch latest data
+    const now: UnixTime = UnixTime.now();
+    const thresholdTime: UnixTime = now.add(
+      -this.priceIntervalMinutes,
+      "minutes",
+    );
+
+    const from: UnixTime = latestBoundary ?? now.add(-30, "days");
+    if (from.toDate().getTime() < thresholdTime.toDate().getTime()) {
+      this.logger.info(
+        `Updating prices for ${asset.coincap_asset_id} from ${from.toDate()} to ${now.toDate()}`,
+      );
+      await this.fetchAndSavePrices(asset, from, now);
+    }
+  }
+
+  private async fetchAndSavePrices(
+    asset: WhitelistedAsset,
+    from: UnixTime,
+    to: UnixTime,
+  ) {
+    const batchSize =
+      (this.apiIntervalLimitMinutes * 24 * 60 * 60 * 1000) /
+      (this.priceIntervalMinutes * 60 * 1000);
+
+    const batchDurationInMs =
```

Comment:
So esentially, `batchDurationInMs = this.apiIntervalLimitMinutes * 24 * 60 * 60 * 1000`. I understand where  `60` and `1000` at the end come from, but `* 24 * 60` at the start make it look like API internval is specified in days, not minutes.

Could you please elaborate on this formula?

Reply:
What about making `batchDurationInMs` a static constant?

Reply:
We can just move `batchDurationInMs` calculation to the constructor, assuming other parameters will remain static constants or just won't change after the constructor invocation.

This will make the formula to be computed only once. It won't save as much and performance is not critical at all here, so more of a suggestion.